### PR TITLE
Fix https://github.com/toggl/backend/issues/9213

### PR DESCRIPTION
**Pull request by asasmoyo**
fix https://github.com/toggl/backend/issues/9213

I think the problem was worker wasn't able to process work from queue. So I did:

1. Make sure worker retrieves pipes ordered by priority

2. Increased worker count

3. Added worker logging

### CODE REVIEW

File: auto_sync.go [hunk 1]
```diff
@@ -1,48 +1,56 @@
...
 	wg.Add(workersCount)
 	for i := 0; i < workersCount; i++ {
-		go pipeWorker()
+		go pipeWorker(i)
 	}
 }
 
-func pipeWorker() {
+func pipeWorker(id int) {
+	defer func() {
```
Comment (fdelbos) [inline auto_sync.go]: Not sure when that will be called since the loop never end?

File: auto_sync.go [hunk 2]
```diff
@@ -55,14 +63,18 @@ func runPipesStub() {
 func pipeWorkerStub() {
 	ranCount := 0
 	gotCount := 0
+	defer func() {
```
Comment (fdelbos) [inline auto_sync.go]: same here

File: auto_sync.go [hunk 3]
```diff
@@ -1,48 +1,56 @@
...
 	for {
 		pipes, err := getPipesFromQueue()
 		if err != nil {
 			bugsnag.Notify(err)
-			break
+			continue
 		}
 		if pipes == nil {
-			break
+			continue
```
Comment (fdelbos) [inline auto_sync.go]: if the error is nil and the pipes is nil it means we are done. Why continue ?

Comment (asasmoyo) [reply auto_sync.go]: oh yes you are right. but the problem with `break` is that there's possibility there will be less number of workers available when next work comes. i think `continue` with sleep for sometime ex: somewhere between `sleepMin` and `sleepMax` would be better. wdyt?

**Comment by supermalf** [comment]
@asasmoyo Can you fix the conflicts, please?

File: auto_sync.go (hunk 3 above)
Comment (fdelbos) [reply auto_sync.go]: All right lets add a sleep for a little bit, i dont think it can hurt

**Comment by asasmoyo** [comment]
@fdelbos please take a look again

**Pull request merged**